parfive/downloader.py

```diff
@@ -0,0 +1,444 @@
+import asyncio
+import contextlib
+import urllib.parse
+from functools import partial
+from concurrent.futures import ThreadPoolExecutor
+
+import aiohttp
+from tqdm import tqdm, tqdm_notebook
+
+from .results import Results
+from .utils import (FailedDownload, Token, default_name, in_notebook,
+                    run_in_thread, get_ftp_size, get_http_size, get_filepath)
+
+try:
+    import aioftp
+except ImportError:
+    aioftp = None
+
+
+__all__ = ['Downloader']
+
+
+class Downloader:
+    """
+    Download files in parallel.
+
+    Parameters
+    ----------
+
+    max_conn : `int`, optional
```

Comment:
Isn't this a bit misleading? Suppose we want to downloading by pieces in the future, then `max_conn` might mean the number of max concurrent connections to one file (assuming the server supports ranged requests, this is a good way to speed up downloads).

I think something like `max_downloads` or `max_concur_downloads` would be a better suit.